node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/acl/AclMaintainer.java

```diff
@@ -1,99 +1,113 @@
-import java.util.Map;
-import java.util.Optional;
+import java.io.File;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 /**
  * The responsibility of this class is to configure ACLs for all running containers. The ACLs are fetched from the Node
  * repository. Based on those ACLs, iptables commands are created and then executed in each of the containers network
  * namespace.
  * <p>
- * If an ACL cannot be configured (e.g. iptables process execution fails), a rollback is attempted by setting the
- * default policy to ACCEPT which will allow any traffic. The configuration will be retried the next time the
- * maintainer runs.
+ * If an ACL cannot be configured (e.g. iptables process execution fails) we attempted to flush the rules
+ * rendering the firewall open.
  * <p>
- * The ACL maintainer does not handle IPv4 addresses and is thus only intended to configure ACLs for IPv6-only
- * containers (e.g. any container, except node-admin).
+ * The configuration will be retried the next time the maintainer runs.
  *
  * @author mpolden
+ * @author smorgrav
  */
 public class AclMaintainer implements Runnable {
 
     private static final PrefixLogger log = PrefixLogger.getNodeAdminLogger(AclMaintainer.class);
-    private static final String IPTABLES_COMMAND = "ip6tables";
 
     private final DockerOperations dockerOperations;
     private final NodeRepository nodeRepository;
+    private final IPAddresses ipAddresses;
     private final String nodeAdminHostname;
-    private final Map<ContainerName, Acl> containerAcls;
 
-    public AclMaintainer(DockerOperations dockerOperations, NodeRepository nodeRepository,
-                         String nodeAdminHostname) {
+    public AclMaintainer(DockerOperations dockerOperations, NodeRepository nodeRepository, String nodeAdminHostname, IPAddresses ipAddresses) {
         this.dockerOperations = dockerOperations;
         this.nodeRepository = nodeRepository;
+        this.ipAddresses = ipAddresses;
         this.nodeAdminHostname = nodeAdminHostname;
-        this.containerAcls = new HashMap<>();
     }
 
-    private boolean isAclActive(ContainerName containerName, Acl acl) {
-        return Optional.ofNullable(containerAcls.get(containerName))
-                .map(acl::equals)
-                .orElse(false);
-    }
+    private void apply(Container container, Acl acl, IPVersion ipVersion) {
 
-    private void applyAcl(ContainerName containerName, Acl acl) {
-        if (isAclActive(containerName, acl)) {
-            return;
-        }
-        final Command flush = new FlushCommand(Chain.INPUT);
-        final Command rollback = new PolicyCommand(Chain.INPUT, Action.ACCEPT);
+        // 1. Get address
+        Optional<InetAddress> address = ipAddresses.getAddress(container.hostname, ipVersion);
```

Comment:
This should probably throw an exception and the caller needs to know which IP addresses this host should have.

Reply:
The normal case is where does not return an address is when the container does not have a ipv4 address (at least not DNS wise) which is the case for our tenant containers today. We could move that logic to the caller of course.